empyrical/tests/test_stats.py

```diff
@@ -1048,6 +1048,90 @@ def empyrical(self):
         )
 
 
+class Test2DStats(TestCase):
+    """
+    Tests for functions that are capable of outputting a DataFrame.
+    """
+
+    input_one = [np.nan, 0.01322056, 0.03063862, -0.01422057,
+                 -0.00489779, 0.01268925, -0.03357711, 0.01797036]
+    input_two = [0.01846232, 0.00793951, -0.01448395, 0.00422537,
+                 -0.00339611, 0.03756813, 0.0151531, 0.03549769]
+
+    expected_0_one = [0.000000, 0.013221, 0.044264, 0.029414, 0.024372,
+                      0.037371, 0.002539, 0.020555]
+    expected_0_two = [0.018462, 0.026548, 0.011680, 0.015955, 0.012504,
+                      0.050542, 0.066461, 0.104318]
+
+    expected_100_one = [100.000000, 101.322056, 104.426424, 102.941421,
+                        102.437235, 103.737087, 100.253895, 102.055494]
+    expected_100_two = [101.846232, 102.654841, 101.167994, 101.595466,
+                        101.250436, 105.054226, 106.646123, 110.431814]
+
+    df_index = pd.date_range('2000-1-30', periods=8, freq='D')
+
+    df_input = pd.DataFrame({
+        'one': pd.Series(input_one, index=df_index),
+        'two': pd.Series(input_two, index=df_index)})
+
+    df_0_expected = pd.DataFrame({
+        'one': pd.Series(expected_0_one, index=df_index),
+        'two': pd.Series(expected_0_two, index=df_index)})
+
+    df_100_expected = pd.DataFrame({
+        'one': pd.Series(expected_100_one, index=df_index),
+        'two': pd.Series(expected_100_two, index=df_index)})
+
+    @parameterized.expand([
+        (df_input, 0, df_0_expected),
+        (df_input, 100, df_100_expected)
+    ])
+    def test_cum_returns_df(self, returns, starting_value, expected):
+        cum_returns = self.empyrical.cum_returns(
+            returns,
+            starting_value=starting_value,
+        )
+
+        for column in returns.columns:
+            for i in range(len(returns[column])):
+                assert_almost_equal(
+                    cum_returns[column][i],
+                    expected[column][i],
+                    4)
+
+    @parameterized.expand([
+        (df_input.as_matrix(), 0, df_0_expected.as_matrix()),
+        (df_input.as_matrix(), 100, df_100_expected.as_matrix())
+    ])
+    def test_cum_returns_matrix(self, returns, starting_value, expected):
```

Comment:
Instead of duplicating the test methods for the input type, can we reuse the `EmpyricalProxy` machinery to add a subclass?
```python
@property
def empyrical(self):
    return PassArraysEmpyricalProxy(self, np.ndarray)
```

If we have one class that just deals with DataFrames and another with ndarrays, then each can check that the output type matches the input type.  Right now, we're asserting that DataFrame input will come back as either a DataFrame or an ndarray (and same with ndarray).